### PR TITLE
csi: introduce new log level for sidecars

### PR DESCRIPTION
**Pull request by yati1998**
This commit introduces new log level called csidecarLogLevel
for the sidecars which are not maintained by the csi-driver.
This will allow to have different configuration options
for the sidecars like snapshotter, resizer, attacher and
provisioner containers which will help to avoid the
spamming of logs to an extent.

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

**Comment by yati1998** [comment]
@Madhu-1 and @humblec Would like to get your review on this.

File: deploy/charts/rook-ceph/values.yaml [hunk 1]
```diff
@@ -101,9 +101,12 @@ csi:
   # Set replicas for csi provisioner deployment.
   provisionerReplicas: 2
 
-  # Set logging level for csi containers.
+  # Set logging level for csi containers maintained by the CSI driver.
```
Comment (Madhu-1) [inline deploy/charts/rook-ceph/values.yaml:104]: `Set logging level for cephCSI containers maintained by the cephCSI`

File: deploy/charts/rook-ceph/values.yaml [hunk 2]
```diff
@@ -305,22 +308,22 @@ csi:
   # CEPH CSI NFS plugin resource requirement list, Put here list of resource
   # requests and limits you want to apply for plugin pod
   csiNFSPluginResource: |
-   - name : driver-registrar
```
Comment (Madhu-1) [inline deploy/charts/rook-ceph/values.yaml:308]: can you please revert this change?

File: deploy/charts/rook-ceph/values.yaml [hunk 3]
```diff
@@ -347,17 +350,17 @@ csi:
   #rbdLivenessMetricsPort: 9080
   #kubeletDirPath: /var/lib/kubelet
   #cephcsi:
-    #image: quay.io/cephcsi/cephcsi:v3.6.2
```
Comment (Madhu-1) [inline deploy/charts/rook-ceph/values.yaml:350]: same here aswell

File: deploy/examples/operator-openshift.yaml [hunk 4]
```diff
@@ -127,10 +127,14 @@ data:
   # there is significant drop in read/write performance.
   # CSI_ENABLE_HOST_NETWORK: "true"
 
-  # Set logging level for csi containers.
+  # Set logging level for csi containers maintained by the CSI Driver.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   # CSI_LOG_LEVEL: "0"
 
+  # Set logging level for csi containers not maintained by the CSI driver.
```
Comment (Madhu-1) [inline deploy/examples/operator-openshift.yaml:134]: Set logging level for Kubernetes-csi sidecar containers`

File: deploy/examples/operator.yaml [hunk 5]
```diff
@@ -42,10 +42,14 @@ data:
...
   # CSI_ENABLE_HOST_NETWORK: "true"
 
-  # Set logging level for csi containers.
+  # Set logging level for csi containers maintained by the CSI driver.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   # CSI_LOG_LEVEL: "0"
 
+  # Set logging level for csi containers not maintained by the CSI driver.
+  # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
+  # CSI_SIDECAR_LOG_LEVEL: "1"
```
Comment (Madhu-1) [inline deploy/examples/operator.yaml:51]: it should be 0?

File: pkg/operator/ceph/csi/spec.go [hunk 6]
```diff
@@ -213,7 +214,8 @@ const (
 
 	detectCSIVersionName = "rook-ceph-csi-detect-version"
 	// default log level for csi containers
-	defaultLogLevel uint8 = 0
+	defaultLogLevel        uint8 = 0
+	defaultSidecarLogLevel uint8 = 1
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/spec.go:218]: default to `0`

File: deploy/charts/rook-ceph/values.yaml [hunk 7]
```diff
@@ -101,9 +101,12 @@ csi:
   # Set replicas for csi provisioner deployment.
   provisionerReplicas: 2
 
-  # Set logging level for csi containers.
+  # Set logging level for csi containers maintained by the CSI driver.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   #logLevel: 0
+  # Set logging level for csi containers not maintained by the CSI driver.
+  # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
+  # sidecarLogLevel: 0
```
Comment (Madhu-1) [inline deploy/charts/rook-ceph/values.yaml:109]: update helm documentation for this one.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: deploy/charts/rook-ceph/values.yaml (hunk 2 above)
Comment (yati1998) [reply deploy/charts/rook-ceph/values.yaml:308]: not sure, tried to do that manually, but on saving, it gets back to this, not sure why these changes occurred. Will check this.

File: deploy/examples/operator-openshift.yaml [hunk 8]
```diff
@@ -127,10 +127,14 @@ data:
...
   # CSI_ENABLE_HOST_NETWORK: "true"
 
-  # Set logging level for csi containers.
+  # Set logging level for Kubernetes-csi sidecar containers.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   # CSI_LOG_LEVEL: "0"
 
+  # Set logging level for cephCSI containers maintained by the cephCSI.
+  # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
+  # CSI_SIDECAR_LOG_LEVEL: "0"
```
Comment (humblec) [inline deploy/examples/operator-openshift.yaml:136]: you can default to `1`

File: deploy/examples/operator.yaml [hunk 9]
```diff
@@ -42,10 +42,14 @@ data:
...
   # CSI_ENABLE_HOST_NETWORK: "true"
 
-  # Set logging level for csi containers.
+  # Set logging level for Kubernetes-csi sidecar containers.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   # CSI_LOG_LEVEL: "0"
 
+  # Set logging level for cephCSI containers maintained by the cephCSI.
+  # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
+  # CSI_SIDECAR_LOG_LEVEL: "0"
```
Comment (humblec) [inline deploy/examples/operator.yaml:51]: here as well

**Review by humblec — COMMENTED** [review]
sidecar log level can be defaulted to `1`

File: Documentation/Helm-Charts/operator-chart.md [hunk 10]
```diff
@@ -74,6 +74,7 @@ The following tables lists the configurable parameters of the rook-operator char
 | `csi.cephFSFSGroupPolicy`           | Policy for modifying a volume's ownership or permissions when the CephFS PVC is being mounted                                                    | `ReadWriteOnceWithFSType`                                 |
 | `csi.nfsFSGroupPolicy`              | Policy for modifying a volume's ownership or permissions when the NFS PVC is being mounted                                                       | `ReadWriteOnceWithFSType`                                 |
 | `csi.logLevel`                      | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                      | `0`                                                       |
+| `csi.sidecarLogLevel`                      | Set logging level for cephCSI containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                      | `0`                                                       |
```
Comment (humblec) [inline Documentation/Helm-Charts/operator-chart.md:77]: s/0/1

**Comment by humblec** [comment]
As discussed the other sidecars here defaulted to `0`, so we could start from `0` for now.. @yati1998

File: deploy/charts/rook-ceph/values.yaml [hunk 11]
```diff
@@ -101,9 +101,12 @@ csi:
   # Set replicas for csi provisioner deployment.
   provisionerReplicas: 2
 
-  # Set logging level for csi containers.
+  # Set logging level for Kubernetes-csi sidecar containers.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   #logLevel: 0
+  # Set logging level for cephCSI containers maintained by the cephCSI.
+  # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
```
Comment (travisn) [inline deploy/charts/rook-ceph/values.yaml:108]:
```suggestion
  # Supported values from 0 to 5. 0 for general useful logs (the default), 5 for trace level verbosity.
```

File: Documentation/Helm-Charts/operator-chart.md (hunk 10 above)
Comment (travisn) [inline Documentation/Helm-Charts/operator-chart.md:77]:
```suggestion
| `csi.sidecarLogLevel`                      | Set logging level for cephCSI containers. Supported values from 0 to 5. 0 for general useful logs (the default), 5 for trace level verbosity.                      | `0`                                                       |
```

File: pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml [hunk 12]
```diff
@@ -25,7 +25,7 @@ spec:
         - name: csi-attacher
           image: {{ .AttacherImage }}
           args:
-            - "--v={{ .LogLevel }}"
+            - "--v={{ .SidecarLogLevel }}"
```
Comment (travisn) [inline pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml:28]: Would we need granularity for each sidecar log level? Maybe it's not worth it, but at the same time it may not make sense to turn up verbosity on all sidecars if you are only troubleshooting one of them.

**Review by travisn — CHANGES_REQUESTED** [review]

File: pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml (hunk 12 above)
Comment (Madhu-1) [reply pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml:28]: I think it should be fine to keep it for all sidecars as this will be turned on/off for a short period of debugging. if we add per sidecar, it might be more complicated and not worthy as you said.

**Review by humblec — APPROVED** [review]

**Comment by humblec** [comment]
@yati1998 its good to add the reference to CSI PR in the PR description.

File: deploy/examples/operator-openshift.yaml [hunk 13]
```diff
@@ -127,10 +127,14 @@ data:
   # there is significant drop in read/write performance.
   # CSI_ENABLE_HOST_NETWORK: "true"
 
-  # Set logging level for csi containers.
+  # Set logging level for Kubernetes-csi sidecar containers.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   # CSI_LOG_LEVEL: "0"
 
+  # Set logging level for cephCSI containers maintained by the cephCSI.
```
Comment (travisn) [inline deploy/examples/operator-openshift.yaml:134]: Is this comment switched with line 130? This description is reversed from my understanding.

File: deploy/examples/operator.yaml [hunk 14]
```diff
@@ -42,10 +42,14 @@ data:
   # there is significant drop in read/write performance.
   # CSI_ENABLE_HOST_NETWORK: "true"
 
-  # Set logging level for csi containers.
+  # Set logging level for Kubernetes-csi sidecar containers.
   # Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.
   # CSI_LOG_LEVEL: "0"
 
+  # Set logging level for cephCSI containers maintained by the cephCSI.
```
Comment (travisn) [inline deploy/examples/operator.yaml:49]: Switched with line 45?

File: pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml (hunk 12 above)
Comment (travisn) [reply pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml:28]: Agreed, let's go with this simple setting for now.

**Review by travisn — CHANGES_REQUESTED** [review]

File: deploy/examples/operator-openshift.yaml (hunk 13 above)
Comment (yati1998) [reply deploy/examples/operator-openshift.yaml:134]: ahh yeah true, Thanks. Updating now.

File: Documentation/Helm-Charts/operator-chart.md [hunk 15]
```diff
@@ -74,6 +74,7 @@ The following tables lists the configurable parameters of the rook-operator char
 | `csi.cephFSFSGroupPolicy`           | Policy for modifying a volume's ownership or permissions when the CephFS PVC is being mounted                                                    | `ReadWriteOnceWithFSType`                                 |
 | `csi.nfsFSGroupPolicy`              | Policy for modifying a volume's ownership or permissions when the NFS PVC is being mounted                                                       | `ReadWriteOnceWithFSType`                                 |
 | `csi.logLevel`                      | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                      | `0`                                                       |
+| `csi.sidecarLogLevel`                      | Set logging level for cephCSI containers. Supported values from 0 to 5. 0 for general useful logs (the default), 5 for trace level verbosity.                      | `0`                                                       |
```
Comment (Madhu-1) [inline Documentation/Helm-Charts/operator-chart.md:77]: `Set logging level for cephCSI containers` to `Set logging level for cephCSI sidecar containers`

File: deploy/charts/rook-ceph/values.yaml [hunk 16]
```diff
@@ -101,9 +101,12 @@ csi:
   # Set replicas for csi provisioner deployment.
   provisionerReplicas: 2
 
-  # Set logging level for csi containers.
+  # Set logging level for Kubernetes-csi sidecar containers.
```
Comment (Madhu-1) [inline deploy/charts/rook-ceph/values.yaml:104]: `logLevel` is for cephcsi containers and `sidecarLogLevel` is for sidecar containers, right? but comment is not matching the expectation

File: Documentation/Helm-Charts/operator-chart.md [hunk 17]
```diff
@@ -74,6 +74,7 @@ The following tables lists the configurable parameters of the rook-operator char
 | `csi.cephFSFSGroupPolicy`           | Policy for modifying a volume's ownership or permissions when the CephFS PVC is being mounted                                                    | `ReadWriteOnceWithFSType`                                 |
 | `csi.nfsFSGroupPolicy`              | Policy for modifying a volume's ownership or permissions when the NFS PVC is being mounted                                                       | `ReadWriteOnceWithFSType`                                 |
 | `csi.logLevel`                      | Set logging level for csi containers. Supported values from 0 to 5. 0 for general useful logs, 5 for trace level verbosity.                      | `0`                                                       |
```
Comment (Madhu-1) [inline Documentation/Helm-Charts/operator-chart.md:76]: update comment for `csi.logLevel` to `cephCSI containers maintained by the cephCSI` to reflect it only applicable for cephcsi containers

**Review by Madhu-1 — APPROVED** [review]

File: Documentation/Helm-Charts/operator-chart.md (hunk 17 above)
Comment (yati1998) [reply Documentation/Helm-Charts/operator-chart.md]: done

**Review by travisn — APPROVED** [review]
